src/chunk.c

```diff
@@ -3404,7 +3404,7 @@ ts_chunk_drop_chunks(PG_FUNCTION_ARGS)
 	Dimension *time_dim;
 	Oid time_type;
 
-	PreventCommandIfReadOnly(DROP_CHUNKS_FUNCNAME "()");
+	PreventCommandIfReadOnly(psprintf("%s()", get_func_name(FC_FN_OID(fcinfo))));
```

Comment:
Can we create a nicer macro/function for this, e.g.:

```
#define TsFuncPreventCommandIfReadOnly PreventCommandIfReadOnly(psprintf("%s()", get_func_name(FC_FN_OID(fcinfo))));
```

Then we only need to do:

```
TsFuncPreventCommandIfReadOnly;
```

Alternatively pass in `fcinfo`:


```
TsFuncPreventCommandIfReadOnly(fcinfo);
```

Reply:
Replaced it with a macro that also avoids executing the function unless it has to.